### PR TITLE
TE-1509 Expose srset and sizes on all components using ResponsiveImage

### PR DESCRIPTION
**Pull request by Jonnygwi**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1509)

### What **one** thing does this PR do?
Exposes additional props on all components using ResponsiveImage so that we can utilise srcset and sizes.

### Any other notes
- Refactors Thumbnail to use ResponsiveImage instead of `backgroundImage`.
- Refactors numerous components to use snapshot testing.
- Images below are just showing that the image is successfully being shown. Ignore UI changes that aren't affecting the image as that's due to a current issue with responsive component helpers (ShowOnMobile and ShowOnDesktop).

| Component | Before | After |
| ------------ |-------|-------|
| Dropdown |image | image |
| Hero | image | image |
| FullBleed | image |  image |
| PropertyPageHero | image | image |
| Gallery | image | image |
| Slideshow | image | image |
| Thumbnail | image | image |
| FeaturedProperty | image | image |
| FeaturedProperties | image | image |
| FeaturedRoomType | image | image |
| FeaturedRoomTypes | image | image |
| HomepageHero | image | image |
| Promotion | image  | image |
| Availability | image | image |
| Pictures | image | image |
| Rates | image | image |
| RoomType | image | image |
| RoomTypes | image |image |

### CODE REVIEW

File: src/components/inputs/Dropdown/utils/getHasImages.js [hunk 1]
```diff
@@ -3,4 +3,4 @@
  * @return {boolean}
  */
 export const getHasImages = options =>
-  options.some(option => option.hasOwnProperty('image'));
+  options.some(option => option.hasOwnProperty('imageUrl'));
```
Comment (marcdavi-es) [inline src/components/inputs/Dropdown/utils/getHasImages.js]: This check needs to be broadened to include the case where an option has `imageSrcSet` and `imageSizes`.

File: src/components/property-page-widgets/Rates/__snapshots__/component.spec.js.snap [hunk 2]
```diff
@@ -2328,22 +2328,30 @@ exports[`<Rates /> if \`props.roomTypes\` is passed should have the right struct
   roomTypes={
     Array [
       Object {
-        "image": "🎅",
+        "image": Object {
+          "url": "🎅",
+        },
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Rates/__snapshots__/component.spec.js.snap]: Shouldn't this be

```js
"imageUrl: "🎅","
```

?

File: src/components/property-page-widgets/PropertyPageHero/component.js [hunk 3]
```diff
@@ -36,46 +36,58 @@ const Component = ({
...
-            </Button>
-          }
-        />
-      </HorizontalGutters>
-    </FlexContainer>
-    <Divider />
-  </Hero>
-);
+}) => {
+  const firstImage = images[0];
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/PropertyPageHero/component.js]: No need for this intermediary variable

File: src/components/property-page-widgets/PropertyPageHero/component.spec.js [hunk 4]
```diff
@@ -1,28 +1,18 @@
...
+import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';
 
 import { ComponentWithResponsive as PropertyPageHero } from './component';
 
 const imageUrl = '🚞';
 
 const props = {
   activeNavigationItemIndex: 1,
-  backgroundImageUrl: 'url',
+  backgroundImage: {
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/PropertyPageHero/component.spec.js]: I don't think `PropertyPageHero` accepts `props.backgroundImage`?

These changes need to be made to the items in the `images` array.

File: src/components/media/Slideshow/component.js [hunk 5]
```diff
@@ -63,18 +63,14 @@ Component.propTypes = {
     PropTypes.shape({
       /** Alternative text to show if the image can't be loaded by the browser. */
       alternativeText: PropTypes.string,
-      /** A set of media conditions indicating to the browser which source to choose.
-       *  [See this for more info](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images)
-       */
+      /** URL pointing to the image to display. */
+      imageUrl: PropTypes.string.isRequired,
```
Comment (marcdavi-es) [inline src/components/media/Slideshow/component.js]: In this case I don't think we should prepend `image` - it's already namespaced by being in an `images` array.

File: src/components/media/Slideshow/Readme.md [hunk 6]
```diff
@@ -1,18 +1,18 @@
 ```jsx
 const images = [
   {
-    alternativeText: 'Two cats',
-    url: 'https://li3.cdbcdn.com/oh/522a12d9-ab51-4635-94c1-42536f286e4d.jpg',
+    imageAlternativeText: 'Two cats',
```
Comment (marcdavi-es) [inline src/components/media/Slideshow/Readme.md]: This doesn't seem right? Should be just `alternativeText`, right?

File: src/components/inputs/Dropdown/utils/getHasImages.spec.js [hunk 7]
```diff
@@ -1,17 +1,17 @@
...
+      { value: 'firstValue', imageUrl: '/yoyo/' },
       { value: 'secondValue' },
     ];
     const actual = getHasImages(options);
 
     expect(actual).toBe(true);
   });
 
-  it('should return `false` if object doesnt have `image` property', () => {
+  it('should return `false` if object doesnt have `imageUrl` property', () => {
```
Comment (red-runner) [inline src/components/inputs/Dropdown/utils/getHasImages.spec.js]: `if object doesnt have`  => `if object does not have`

**Pull request merged**